src/main/java/net/imagej/ops/deconvolve/RichardsonLucyNonCirculantC.java

```diff
@@ -0,0 +1,338 @@
+		RandomAccessibleInterval<K> kernel, RandomAccessibleInterval<O> out)
+	{
+
+		// when doing non circulant deconvolution we need to crop and copy back to
+		// the
+		// original image size
+
+		long[] start = new long[k.numDimensions()];
+		long[] end = new long[k.numDimensions()];
+
+		for (int d = 0; d < k.numDimensions(); d++) {
+			start[d] = (getRAIExtendedEstimate().dimension(d) - k.dimension(d)) / 2;
+			end[d] = start[d] + k.dimension(d) - 1;
+		}
+
+		RandomAccessibleInterval<O> temp = ops().image().crop(
+			getRAIExtendedEstimate(), new FinalInterval(start, end));
+
+		ops().copy().rai(out, temp);
+
+	}
+
+	/**
+	 * create the normalization image needed for semi noncirculant model see
+	 * http://bigwww.epfl.ch/deconvolution/challenge2013/index.html?p=doc_math_rl
+	 */
+	protected void createNormalizationImageSemiNonCirculant(
+		RandomAccessibleInterval<K> kernel)
+	{
+
+		// k is the window size (valid image region)
+		int length = k.numDimensions();
+
+		long[] n = new long[length];
+		long[] nFFT = new long[length];
+
+		// n is the valid image size plus the extended region
+		// also referred to as object space size
+		for (int d = 0; d < length; d++) {
+			n[d] = k.dimension(d) + l.dimension(d) - 1;
+		}
+
+		for (int d = 0; d < length; d++) {
+			nFFT[d] = getRAIExtendedReblurred().dimension(d);
+		}
+
+		// create the normalization image
+		final O type = Util.getTypeFromInterval(getRAIExtendedReblurred());
+		normalization = getImgFactory().create(getRAIExtendedReblurred(), type);
+
+		// size of the measurement window
+		Point size = new Point(length);
+		long[] sizel = new long[length];
+
+		for (int d = 0; d < length; d++) {
+			size.setPosition(k.dimension(d), d);
+			sizel[d] = k.dimension(d);
+		}
+
+		// starting point of the measurement window when it is centered in fft space
+		Point start = new Point(length);
+		long[] startl = new long[length];
+		long[] endl = new long[length];
+
+		for (int d = 0; d < length; d++) {
+			start.setPosition((nFFT[d] - k.dimension(d)) / 2, d);
+			startl[d] = (nFFT[d] - k.dimension(d)) / 2;
+			endl[d] = startl[d] + sizel[d] - 1;
+		}
+
+		// size of the object space
+		Point maskSize = new Point(length);
+		long[] maskSizel = new long[length];
+
+		for (int d = 0; d < length; d++) {
+			maskSize.setPosition(Math.min(n[d], nFFT[d]), d);
+			maskSizel[d] = Math.min(n[d], nFFT[d]);
+		}
+
+		// starting point of the object space within the fft space
+		Point maskStart = new Point(length);
+		long[] maskStartl = new long[length];
+
+		for (int d = 0; d < length; d++) {
+			maskStart.setPosition((Math.max(0, nFFT[d] - n[d]) / 2), d);
+			maskStartl[d] = (Math.max(0, nFFT[d] - n[d]) / 2);
+		}
+
+		RandomAccessibleInterval<O> temp = Views.interval(normalization,
+			new FinalInterval(startl, endl));
+		Cursor<O> normCursor = Views.iterable(temp).cursor();
+
+		// draw a cube the size of the measurement space
+		while (normCursor.hasNext()) {
```

Comment:
I think we do not have such op right now, but it will not be hard to create one. I can make a PR later. Which namespace do you think it should go? @dietzc